index.bs

```diff
@@ -4241,19 +4242,26 @@ promises returned by its <a lt="writable stream writer">writer</a>'s {{WritableS
       },
 
       close() {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(1000);
-        });
+        return closeWS(1000);
       },
 
       abort(reason) {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(4000, reason && reason.message);
-        });
-      }
+        return closeWS(4000, reason && reason.message);
+      },
     });
+
+    function closeWS(code, reasonString) {
+      return new Promise((resolve, reject) => {
+        ws.onclose = e => {
```

Comment:
Chrome definitely sets wasClean to false. 

For me the most important sentence is this from RFC6455 section 7.1.7:

   An endpoint MUST NOT continue to attempt to
   process data (including a responding Close frame) from the remote
   endpoint after being instructed to _Fail the WebSocket Connection_.

If we're prohibited from processing the Close frame from the server, there's no way to complete the closing handshake, and so it can't ever be clean.

This is actually quite subtle and it's possible other implementations have read the spec differently. There are a bunch of web-platform-tests where the test expects wasClean = true but Chrome sets it to false, but in my brief investigation it appears this is an interoperability issue with the test server, rather than being caused by any subtle issues.

I think from a standard's point of view, using wasClean in this way should totally work. For compatibility, you might want to have an additional wasOnErrorCalled flag. Personally I think the example should aim for brevity rather than maximum compatibility.